src/main/java/com/google/sps/servlets/InfoCardServlet.java

```diff
@@ -59,20 +72,77 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       return;
     }
 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity chosenElection = ServletHelper.findElectionInDatastore(datastore, electionId);
+
+    if (chosenElection == null) {
+      response.setContentType("text/html");
+      response
+          .getWriter()
+          .println(String.format("Could not find election with ID %s in Datastore.", electionId));
+      response.setStatus(400);
+      return;
+    }
+
+    // TODO: if this election is already populated, we don't need to make another query
+
     URL url =
         new URL(
             String.format(
+                "%s?address=%s&electionId=%s&key=%s",
                 BASE_URL,
                 address,
                 electionId,
                 ServletHelper.getApiKey("112408856470", "election-api-key", "1")));
-    JSONObject obj = ServletHelper.readFromApiUrl(url, response);
+    JSONObject electionListFromApi = ServletHelper.readFromApiUrl(url);
+    JSONArray positionListFromApi = electionListFromApi.getJSONArray("contests");
+    chosenElection.setProperty("positions", fillPositions(datastore, positionListFromApi));
+  }
+
+  /**
+   * Puts candidate Position Entities in the Datastore.
+   *
+   * @param datastore the Datastore containing all election data
+   * @param positionData API-returned list of positions on the election ballot
+   * @return list of Key names of all Position Entities added
+   */
+  private List<String> fillPositions(DatastoreService datastore, JSONArray positionData) {
+    List<String> positionKeyList = new ArrayList<String>();
+    for (Object positionObject : positionData) {
+      JSONObject position = (JSONObject) positionObject;
+
+      Entity positionEntity = new Entity("Position");
+      positionEntity.setProperty("name", position.getString("office"));
+      positionEntity.setProperty(
+          "candidates", fillCandidates(datastore, position.getJSONArray("candidates")));
+      datastore.put(positionEntity);
+
+      positionKeyList.add(positionEntity.getKey().getName());
+    }
+
+    return positionKeyList;
+  }
 
-    // TODO(anooshree, caseyprice): process JSON objects from API to store as
-    //                              Datastore entities using decomposed functions
+  /**
+   * Puts Candidate Entities in the Datastore.
+   *
+   * @param datastore the Datastore containing all election data
+   * @param candidateData API-returned list of candidates for a position on the election ballot
+   * @return list of Key names of all Candidate Entities added
+   */
+  private List<String> fillCandidates(DatastoreService datastore, JSONArray candidateData) {
```

Comment:
Consider renaming this function. "fillCandidates" implies that you're populating some object. Don't be afraid of long but descriptive function names, especially in Java. 

populateDatastoreWithCandidateEntities()?